### PR TITLE
Launch next-gen correlator tasks

### PR DESCRIPTION
**Pull request by bmerry**
Closes NGC-190.

### CODE REVIEW

**Review by james-smith-za — CHANGES_REQUESTED** [review]
Mainly related to the merge/rename of the modules.

**Comment by bmerry** [comment]
I'm going to suggest that for this PR I stick with the "katfgpu" and "katxgpu" Docker images. They're in the Docker registry, so this will continue to work but just use those old versions (which at least are known to work right now).

When I switch to using katgpucbf there will be a few other changes e.g. to wire up the katcp port on fgpu. Rather than continually trying to keep this PR up to date with the refactoring on katgpucbf I'd like to get this merged and then start new PRs for updates.

I've only glanced at the other comments, but they sound reasonable. Will come back to them later, probably next week.

**Comment by james-smith-za** [comment]
> 
> 
> I'm going to suggest that for this PR I stick with the "katfgpu" and "katxgpu" Docker images. They're in the Docker registry, so this will continue to work but just use those old versions (which at least are known to work right now).
> 
> When I switch to using katgpucbf there will be a few other changes e.g. to wire up the katcp port on fgpu. Rather than continually trying to keep this PR up to date with the refactoring on katgpucbf I'd like to get this merged and then start new PRs for updates.
> 
That's quite reasonable. I'm good with that. The remaining stuff was very minor.

File: katsdpcontroller/generator.py [hunk 1]
```diff
@@ -249,6 +255,221 @@ def _make_meta_writer(g: networkx.MultiDiGraph,
...
+            # Enable cap_net_raw capability for access to raw QPs
+            xgpu.capabilities.append('NET_RAW')
+            # Use the core number as completion vector. This ensures that
+            # multiple instances on a machine will use distinct vectors.
+            xgpu.command += [
+                '--receiver-comp-vector-affinity', '{cores[core]}'
+            ]
+        xgpu.command += stream.command_line_extra
+        # xgpu doesn't use katsdpservices or telstate
+        xgpu.katsdpservices_logging = False
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpcontroller/generator.py:450]: Indeed.

File: katsdpcontroller/scheduler.py [hunk 2]
```diff
@@ -727,17 +727,24 @@ class GPURequest(ResourceRequestsContainer):
         cores (ignored if no CPU cores are reserved).
     name : str, optional
         If specified, the name of the GPU must match this.
+    min_compute_capability : Tuple[int], optional
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpcontroller/scheduler.py:730]: You're quite right. I might have been intending of `Tuple[int, ...]` (variable-length tuple of ints), but compute capabilities are always two components so I'll change it.

File: katsdpcontroller/product_config.py [hunk 3]
```diff
@@ -324,6 +324,17 @@ def __init__(self, name: str, src_streams: Sequence[Stream], *,
         self.antenna = antenna
         self.command_line_extra = list(command_line_extra)
 
+    @property
+    def bits_per_sample(self) -> int:
+        return 10
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpcontroller/product_config.py:329]: I'm not sure - have changed it as suggested.

**Review by james-smith-za — APPROVED** [review]

**Pull request merged**